Shared/ReflectionHelper.cs

```diff
@@ -882,7 +882,16 @@ public static Action Bind<T>(this Action del, Action<T> f, T v)
             del += () => f(v); 
             return del;
         }
-
+        
+        /// <summary> Clones a object. creates a new serializer if needed, but reuses an existing one if possible.</summary>
+        public static T Clone<T>(T obj, ref TapSerializer cachedSerializer)
+        {
+            if (obj is ICloneable cloneable)
+                return (T)cloneable.Clone();
+            
+            return (T)(cachedSerializer ??= new TapSerializer()).Clone(obj);
```

Comment:
This cached serializer is a bit crazy. Could the caller be responsible for providing a (potentially cached) serializer instead?